nixos/modules/services/misc/geoip-databases.nix

```diff
@@ -0,0 +1,166 @@
+        done
+        return $ret
+    }
+    unpackDatabases()
+    {
+        (cd "${cfg.databaseDir}" &&
+            for f in *.gz; do
+                test -f "$f" || continue
+                "${pkgs.gzip}/bin/gzip" --decompress --force -v "$f"
+            done;
+            for f in *.xz; do
+                test -f "$f" || continue
+                "${pkgs.xz.bin}/bin/xz" --decompress --force -v "$f"
+            done
+        )
+    }
+    waitNetworkOnline()
+    {
+        ret=1
+        url=http://geolite.maxmind.com/
+        for i in $(seq 6); do
+            printf "Checking network connectivity, try $i ..."
+            "${pkgs.curl.bin}/bin/curl" --silent --show-error "$url"
+            if [ $? -eq 0 ]; then
+                printf "ok\n"
+                ret=0
+                break
+            fi
+            sleep 10
+        done
+        return $ret
+    }
+    waitNetworkOnline
+    if [ $? -ne 0 ]; then
+        echo "Network is down."
+        exit 1
+    fi
+    echo "Updating GeoIP databases in ${cfg.databaseDir} ..."
+    err_msg=""
+    fetchDatabases || err_msg=", with error(s)"
+    unpackDatabases || err_msg=", with error(s)"
+    echo "...completed GeoIP database update in ${cfg.databaseDir}$err_msg"
+    if [ "$err_msg" != "" ]; then
+        false  # make the overall service fail
+    fi
+  '';
+
+in
+
+{
+  options = {
+    services.geoip-databases = {
+      enable = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          Whether to enable periodic downloading of GeoIP Legacy databases from
+          maxmind.com. You might want to enable this if you, for instance, use
+          ntopng or Wireshark.
+        '';
+      };
+
+      interval = mkOption {
+        type = types.str;
+        default = "weekly";
+        description = ''
+          Update the GeoIP databases at this time / interval.
+          The format is described in
+          <citerefentry><refentrytitle>systemd.time</refentrytitle>
+          <manvolnum>7</manvolnum></citerefentry>.
+        '';
+      };
+
+      databaseDir = mkOption {
+        default = "/var/lib/geoip-databases";
+        description = ''
+          Directory that will contain GeoIP databases.
+        '';
+      };
+
+      databases = mkOption {
+        default = [
+          "GeoLite2-Country.mmdb.gz"
+          "GeoLiteCountry/GeoIP.dat.gz"
+          "GeoIPv6.dat.gz"
+          "GeoLiteCity.dat.xz"
+          "GeoLiteCityv6-beta/GeoLiteCityv6.dat.gz"
+          "asnum/GeoIPASNum.dat.gz"
+          "asnum/GeoIPASNumv6.dat.gz"
+        ];
+        description = ''
+          Which GeoIP databases to update.
+        '';
+      };
+
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+
+    systemd.timers.geoip-updater =
+      { description = "GeoIP Updater Service Timer";
+        partOf      = [ "geoip-updater.service" ];
+        wantedBy    = [ "timers.target" ];
+        timerConfig.OnCalendar = cfg.interval;
```

Comment:
to avoid peaks on maxmind's server backend, we should use a random delay here:

```
RandomizedDelaySec=3600
```

or an Interval between updates instead of fixed dates:

```
OnBootSec=15min
OnUnitActiveSec=1w
```